mapie/regression/utils/ccp_phi_function.py

```diff
@@ -0,0 +1,974 @@
+
+    Parameters
+    ----------
+    functions: Optional[Union[Callable, Iterable[Callable]]]
+        List of functions (or PhiFunction objects) or single function.
+        Each function can take a combinaison of the following arguments:
+        - ``X``: Input dataset, of shape (n_samples, ``n_in``)
+        - ``y_pred``: estimator prediction, of shape (n_samples,)
+        - ``z``: exogenous variable, of shape (n_samples, n_features).
+            It should be given in the ``fit`` and ``predict`` methods.
+        The results of each functions will be concatenated to build the final
+        result of the phi function, of shape (n_samples, ``n_out``).
+        If ``None``, the resulting phi object will return a column of ones,
+        when called. It will result, in the MapieCCPRegressor, in a basic
+        split CP approach.
+
+        By default ``None``.
+
+    marginal_guarantee: bool
+        Add a column of ones to the features, for safety reason
+        (to garanty the marginal coverage, no matter how the other features
+        the ``PhiFunction``object were built).
+        If the ``PhiFunction``object definition covers all the dataset
+        (meaning, for all calibration and test samples, ``phi(X, y_pred, z)``
+        is never all zeros), this column of ones is not necessary
+        to obtain marginal coverage.
+        In this case, you can set this argument to ``False``.
+
+        Note: Even if it is not always necessary to guarantee the marginal
+        coverage, it can't degrade the prediction intervals.
+
+        By default ``True``.
+
+    normalized: bool
+        Whether or not to normalized ``phi(X, y_pred, z)``. Normalization
+        will result in a bounded interval prediction width, avoiding the width
+        to explode to +inf or crash to zero. It is particularly intersting when
+        you know that the conformity scores are bounded. It also prevent the
+        interval to have a interval of zero width for out-of-distribution or
+        new samples. On the opposite, it is not recommended if the conformity
+        scores can vary a lot.
+
+        By default ``False``
+
+    Attributes
+    ----------
+    fit_attributes: Optional[List[str]]
+        Name of attributes set during the ``fit`` method, and required to call
+        ``transform``. Empty list will result in a ``PhiFunction`` class which
+        is always fittedd (doesn't need to be fitted). If ``None``, it will
+        check for a method ``__sklearn_is_fitted__()`` or arguments which ends
+        with ``'_'``, to check if it is fitted or not.
+
+    n_in: int
+        Number of features of ``X``
+
+    n_out: int
+        Number of features of phi(``X``, ``y_pred``, ``z``)
+    """
+
+    fit_attributes: Optional[List[str]] = []
+    output_attributes = ["n_in", "n_out", "init_value"]
+
+    def __init__(
+        self,
+        functions: Optional[Union[Callable, Iterable[Callable]]] = None,
+        marginal_guarantee: bool = True,
```

Comment:
Suggestion: replace `marginal_guarantee` by `biais`. Perhaps make the value default to 'False'.

Reply:
Done